engine/lib/analysis_utils.ml

```diff
@@ -0,0 +1,11 @@
+open Base
+
+module type ANALYSIS = sig
+  (* module FA : Features.T *)
+  module A : Ast.T
+
+  type pre_data
+  type analysis_data
+
+  val analyse : pre_data -> A.item list -> analysis_data
+end
```

Comment:
Do we actually need to force analyses to be modules with a specific type?
An analyse could also operate on a different level (i.e. expressions), so maybe it'd be nicer to let analyses be just functions without constraints on their type signatures, wdyt?
Backends or phases can then freely call them in a natural way